src/components/Home/News/ModalSlider/ModalSlider.tsx

```diff
@@ -0,0 +1,218 @@
+import { FC, useEffect, useState } from "react"
+import { View, Pressable, Dimensions, Image, StyleSheet } from "react-native"
+import { usePalette } from "utils/colors"
+import deleteSvg from "assets/modal/delete.svg"
+import { ZoomableImage } from "./ZoomableImage"
+import { Easing, useSharedValue, withTiming } from "react-native-reanimated"
+import arrow_right from "assets/articles/arrow_right.svg"
+import arrow_left from "assets/articles/arrow_left.svg"
+import { BodyText } from "components/Text"
+import { Icon } from "components/Icon"
+import { GestureHandlerRootView } from "react-native-gesture-handler"
+import { Portal } from "react-native-portalize"
+import _Modal from "react-native-modal"
+export interface ModalSliderProps {
+  imageSources: string[]
+  activeIndex: number
+  onClose: () => void
+  isVisible: boolean
+}
+
+/**
+ * Image Slider Component.
+ *
+ * Implements a full screen Modal.
+ */
+export const ModalSlider: FC<ModalSliderProps> = props => {
+  const { modalBarrier } = usePalette()
+
+  const fullWidth = Dimensions.get("window").width
+
+  const fullHeight = Dimensions.get("window").height
+
+  const [imageIndex, setImageIndex] = useState(props.activeIndex)
+
+  const [source, setSource] = useState<string | undefined>(undefined)
+
+  useEffect(() => {
+    setImageIndex(props.activeIndex)
+  }, [props.activeIndex])
+
+  useEffect(() => {
+    if (props.imageSources.length > 0) {
+      setSource(props.imageSources[imageIndex])
+    }
+  }, [imageIndex, props.imageSources])
+
+  const [height, setHeight] = useState(0)
+
+  useEffect(() => {
+    if (props.imageSources.length > 0 && source) {
+      Image.getSize(source, (width, height) => {
+        setHeight((height * fullWidth) / width)
+      })
+    }
+  }, [source])
+
+  const scale = useSharedValue(1)
+  const savedScale = useSharedValue(1)
+  const position = useSharedValue(0)
+  const lastPosition = useSharedValue(0)
+
+  return (
+    <Portal>
+      <_Modal
```

Comment:
per quanto riguarda questo non so, sono un po' combattuto perché ci sono un paio di gesture che sarebbe carino funzionassero, tipo lo swipe a destra e sinistra per andare all'immagine precedente/successiva, però direi che si può fixare in un altro branch quando abbiamo meno cose prioritarie da fare

Reply:
Ok right. Intendi tipo far comparire dal lato l'immagine successiva mentre la corrente viene swipata? O qualcosa tipo fading della successiva sulla corrente?

Reply:
si beh ho notato che ci sono un paio di cose che mi aspetterei:
- pan dell'immagine in verticale, ovviamente da larghezza massima no, ma quando si zooma si dovrebbe poter fare un pan anche in verticale e non solo in orizzontale
- sì lo swipe deve mostrarti l'immagine successiva dal lato in teoria, che boh un casino non so come farlo funzionare con anche il pan
- swipe verso il basso per chiudere la visione full screen
- sarebbe anche carino se si potesse tappare per far scomparire i controlli e far diventare lo sfondo completamente nero per vedere meglio l'immagine